src/main/java/dev/datainmotion/airquality/AirQualityApp.java

```diff
@@ -45,50 +43,36 @@ public static void main(String[] args) {
         SpringApplication.run(AirQualityApp.class, args);
     }
 
-    @Autowired
-    private PulsarTemplate<Observation> pulsarTemplate;
+	@Bean
+	public PulsarTopic newTopic(PulsarProperties pulsarProperties) {
+		return PulsarTopic.builder(pulsarProperties.getProducer().getTopicName()).build();
+	}
 
-    /**
-     * get rows
-     */
-    private void getRows() {
-        pulsarTemplate.setSchema(Schema.JSON(Observation.class));
-        List<Observation> obsList = airQualityService.fetchCurrentObservation();
-
-        if (obsList == null || obsList.size() <= 0) {
+	@Scheduled(initialDelay = 2000, fixedRate = 2000)
+    public void getRows() {
+		this.pulsarTemplate.setSchema(Schema.JSON(Observation.class));
+		List<Observation> observations = airQualityService.fetchCurrentObservation();
+        if (observations == null || observations.size() <= 0) {
             return;
         }
-        log.debug("Count: {}", obsList.size());
-
-        for (Observation observation2 : obsList) {
-            log.info("{}={} for {} {}",
-                    observation2.getParameterName(),
-                    observation2.getAqi(),
-                    observation2.getStateCode(),
-                    observation2.getReportingArea());
-            try {
-                UUID uuidKey = UUID.randomUUID();
-                MessageId msgid = pulsarTemplate.newMessage(observation2)
-                        .withMessageCustomizer((mb) -> mb.key(uuidKey.toString()))
-                        .withTopic(topicName)
-                        .send();
-
-                log.info("Sent {}", observation2.toString());
-                log.debug("PULSAR MSGID {}", msgid.toString());
-            } catch (Throwable e) {
-                e.printStackTrace();
-                log.error("Pulsar Error", e);
-            }
-        }
-    }
-
-    @Scheduled(initialDelay = 0, fixedRate = 2000)
-    public void repeatRun() {
-        getRows();
-    }
-
-    @Override
-    public void run(String... args) {
-        getRows();
+        log.debug("Count: {}", observations.size());
+		observations.forEach((observation) -> {
+			log.info("{}={} for {} {}",
+					observation.getParameterName(),
+					observation.getAqi(),
+					observation.getStateCode(),
+					observation.getReportingArea());
+			try {
+				UUID uuidKey = UUID.randomUUID();
+				MessageId msgid = pulsarTemplate.newMessage(observation)
+						.withMessageCustomizer((mb) -> mb.key(uuidKey.toString()))
+						.send();
+				log.info("Sent {}", observation);
+				log.debug("PULSAR MSGID {}", msgid.toString());
+			}
+			catch (Throwable e) {
```

Comment:
@tspannhw the next cool thing would be to add a `@PulsarListener` that just logs the events sent to the topic.